app/controllers/api/v2/settings_controller.rb

```diff
@@ -9,6 +9,7 @@ class SettingsController < V2::BaseController
 
       def index
         @settings = resource_scope_for_index.live_descendants
+        @settings.map! { |s| Foreman.settings.find(s.name) }
```

Comment:
`@settings` is a relation here, so `map!` probably doesn't work on it:
```suggestion
        @settings = resource_scope_for_index.live_descendants.map { |s| Foreman.settings.find(s.name) }
```
Although, shouldn't we pull the settings from the registry rather than the db here?

Reply:
We should in the future, but I'm trying to keep the changes minimal here and getting it from registry would require implementing the search on the registry or dropping the search, I'd like to do so in a followup PR, when we already have the registry implementation in.

Reply:
I guess I missunderstood this comment, fixed it according to your comment.